File: easy/official/power_of_thor_episode_1/src/main.py
#! /usr/bin/env python3

light_x, light_y, initial_tx, initial_ty = [int(i) for i in input().split()]

while True:
    remaining_turns = int(input())
    direction = ""

    if initial_ty < light_y:
        direction += "S"
        initial_ty += 1
    elif initial_ty > light_y:
        direction += "N"
        initial_ty -= 1

    if initial_tx < light_x:
        direction += "E"
        initial_tx += 1
    elif initial_tx > light_x:
        direction += "W"
        initial_tx -= 1

    print(direction)

File: easy/official/power_of_thor_episode_1/src/main.c
#include <stdio.h>

void solve(int light_x, int light_y, int initial_tx, int initial_ty);

int main() {
    int light_x;
    int light_y;
    int initial_tx;
    int initial_ty;

    scanf("%d%d%d%d", &light_x, &light_y, &initial_tx, &initial_ty);
    solve(light_x, light_y, initial_tx, initial_ty);

    return 0;
}

void solve(int light_x, int light_y, int initial_tx, int initial_ty) {
    while (1) {
        int remianing_turns;
        scanf("%d", &remianing_turns);

        if (initial_ty < light_y) {
            printf("S");
            initial_ty++;
        }

        if (initial_ty > light_y) {
            printf("N");
            initial_ty--;
        }

        if (initial_tx < light_x) {
            printf("E");
            initial_tx++;
        }

        if (initial_tx > light_x) {
            printf("W");
            initial_tx--;
        }

        printf("\n");
    }
}
